frontend/projects/ui/src/app/services/marketplace.service.ts

```diff
@@ -27,33 +30,49 @@ import {
 export class MarketplaceService extends AbstractMarketplaceService {
   private readonly notes = new Map<string, Record<string, string>>()
 
-  private readonly init$: Observable<Marketplace> = this.patch
+  private readonly data$: Observable<UIMarketplaceData> = this.patch
     .watch$('ui', 'marketplace')
-    .pipe(
-      map(marketplace =>
-        marketplace?.['selected-id']
-          ? marketplace['known-hosts'][marketplace['selected-id']]
-          : this.config.marketplace,
-      ),
-      shareReplay(),
-    )
+    .pipe(take(1), shareReplay({ bufferSize: 1, refCount: true }))
```

Comment:
with `bufferSize` and `refCount`, this essentially enables one subscription?

Reply:
This enables auto unsubscribe if all subscribers have unsubscribed.